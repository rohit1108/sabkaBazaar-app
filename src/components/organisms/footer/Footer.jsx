import React from 'react'

import './Footer.scss';
const Footer = () =>{
   

return(
  <footer className="footer">
  <div className="footer__text" data-testid="footer-text">Copyright Â© 2011-2018 Sabka Baazar Gorcery Supplies Pvt. Ltd.</div>
  </footer>
  );
}


export default Footer;