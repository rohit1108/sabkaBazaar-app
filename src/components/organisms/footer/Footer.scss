@import '../../../styles/variables';

.footer {
    min-height: 5vh;
    background-color: $secondary-color;
    display: flex;

    &__text{
        color: $text-color;
        font-size: $text-font-size;
        margin-left: 10%;
        padding-top: 10px;

        @media (max-width: $breakpoint-mobile){
            font-size: 0.7rem;
        }
    }
}