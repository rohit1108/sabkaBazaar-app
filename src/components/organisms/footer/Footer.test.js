import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Footer from './Footer';

describe("Footer Component", () => {

    it("should render the component properly", () => {
        const footerEl = document.createElement("footer");
        ReactDom.render(<Footer/>, footerEl); 
    });

    it("should display footer text correctly", () => {
        render(<Footer/>);
        expect(screen.getByTestId("footer-text")).toHaveTextContent("Copyright Â© 2011-2018 Sabka Bazaar Grocery Supplies Pvt. Ltd.");

    });

    it('should have the right footer component snapshot', () => {
        const tree = renderer.create(<Footer/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});