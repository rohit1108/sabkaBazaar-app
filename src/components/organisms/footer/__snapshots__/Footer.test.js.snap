// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should have the right footer component snapshot 1`] = `
<footer
  className="footer"
  data-testid="footer-text"
>
  <p
    className="footer__text"
  >
    Copyright Â© 2011-2018 Sabka Baazar Gorcery Supplies Pvt. Ltd.
  </p>
</footer>
`;
